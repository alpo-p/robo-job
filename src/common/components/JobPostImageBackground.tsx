import React from 'react'
import { ImageBackground } from 'react-native'

interface P {
  pictureUrl: string
  children: React.ReactNode
}

export default ({ pictureUrl, children }: P) => (
  <ImageBackground
    source={{ uri: pictureUrl }}
    resizeMode="stretch"
    style={{
      height: 280,
    }}
  >
    {children}
  </ImageBackground>
)
