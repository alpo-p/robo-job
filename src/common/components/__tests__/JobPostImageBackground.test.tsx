import React from 'react'
import renderer from 'react-test-renderer'
import { Text } from 'react-native'
import JobPostImageBackground from '../JobPostImageBackground'

describe('JobPostImageBackground', () => {
  it('should match snapshot', () => {
    const tree = renderer.create(
      <JobPostImageBackground pictureUrl="testUrl">
        <Text>Test</Text>
      </JobPostImageBackground>,
    )
    expect(tree).toMatchSnapshot()
  })
})
