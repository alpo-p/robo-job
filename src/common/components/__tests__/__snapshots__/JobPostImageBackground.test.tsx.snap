// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`JobPostImageBackground should match snapshot 1`] = `
<View
  accessibilityIgnoresInvertColors={true}
  style={
    Object {
      "height": 280,
    }
  }
>
  <Image
    resizeMode="stretch"
    source={
      Object {
        "uri": "testUrl",
      }
    }
    style={
      Array [
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        Object {
          "height": 280,
          "width": undefined,
        },
        undefined,
      ]
    }
  />
  <Text>
    Test
  </Text>
</View>
`;
